examples/sample-layers/label-layer/label-layer.js

```diff
@@ -0,0 +1,54 @@
+import {CompositeLayer, IconLayer} from 'deck.gl';
+import {makeTextureAtlasFromLabels} from './label-utils';
+
+/* Constants */
+const defaultProps = {
+  getLabel: x => x.label,
+  fontSize: 24
+};
+
+/*
+ * A layer that plots a surface based on a z=f(x,y) equation.
+ *
+ * @class
+ * @param {Object} [props]
+ * @param {Number} [props.axesOffset] - amount to set back grids from the plot,
+      relative to the size of the bounding box
+ * @param {Number} [props.fontSize] - size of the labels
+ * @param {Array} [props.axesColor] - color of the gridlines, in [r,g,b,a]
+ */
+export default class LabelLayer extends CompositeLayer {
+  updateState({props, oldProps, changeFlags}) {
+    if (changeFlags.dataChanged || props.fontSize !== oldProps.fontSize) {
+      this.updateLabelAtlas(props);
+    }
+  }
+
+  updateLabelAtlas() {
+    const {gl} = this.context;
+    const {data, getLabel, fontSize} = this.props;
+
+    const {texture, mapping} = makeTextureAtlasFromLabels(gl, {data, getLabel, fontSize});
+    // if (this.state.texture) {
+    //   this.state.texture.delete();
+    // }
+    this.setState({texture, mapping});
+  }
+
+  renderLayers() {
+    return new IconLayer(Object.assign({}, this.props, {
```

Comment:
IconLayer doesn't currently preserve aspect ratio. We should fix it in a separate PR.